proposals/004-skips/README.md

```diff
@@ -0,0 +1,179 @@
+    for _, c := range app.Spec.InitContainers {
+        if c.Image == app.Spec.Image {
+            return nil, errors.New("container in app.Spec.InitContainers must have a different image than Spin App")
+        }
+        if c.Name == "" {
+            return nil, errors.New("container in app.Spec.InitContainers must have a name")
+        }
+        if c.Name == app.Name {
+            return nil, errors.New("container in app.Spec.InitContainers must have a different name than the Spin App")
+        }
+        c.Env = append(c.Env, env...)
+        c.VolumeMounts = append(c.VolumeMounts, volumeMounts...)
+        if c.Resources.Limits == nil && c.Resources.Requests == nil {
+            c.Resources = resources
+        }
+        if c.LivenessProbe == nil {
+            c.LivenessProbe = livenessProbe
+        }
+        if c.ReadinessProbe == nil {
+            c.ReadinessProbe = readinessProbe
+        }
+        containers = append(containers, c)
+    }
+}
+```
+
+See the full diff [here](https://github.com/spinkube/spin-operator/compare/main...Mossaka:spin-operator-msk:sidecars)
+
+[sidecar containers]: (https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)
+[InitContainers]: (https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+### Ports
+
+Sidecar containers must avoid using port 80, which is reserved for the Spin application. Other ports like 8080 or 8081 can be used.
+
+### Services
+
+Sidecar containers can communicate internally using `localhost`, but if external access is needed, users must create a separate service for the sidecar container.
+
+### Restrictions
+
+* Sidecar containers cannot use the same image or name as the Spin App. 
+* Sidecar containers must not use port 80.
+
+### Sample SpinApp yaml
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: simple-spinapp
+spec:
+  image: "ttl.sh/spinapp:44h"
+  replicas: 1
+  executor: containerd-shim-spin
+  volumes:
+    - name: example-volume
+      persistentVolumeClaim:
+        claimName: example-pv-claim
+  volumeMounts:
+    - name: example-volume
+      mountPath: "/mnt/data"
+  initContainers:
+    - name: logshipper
+      image: alpine:latest
+      restartPolicy: Always
+      command: ['sh', '-c', 'tail -F /opt/logs.txt']
+      volumeMounts:
+      - name: example-volume
+        mountPath: "/mnt/data"
+```
+
+Note: The `InitContainers` field is intentionally designed to follow the the design of sidecar containers in Kubernetes Pods. You will need to have `SidecarContainers` feature gate enabled in your Kubernetes cluster to use this feature. See more details [here](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/).
+
+### Init Containers
+
+Since the sidecar containers are just a special case of InitContainers, I realized that the scope of this SKIP is naturally expanded to include the support of init containers in the Spin Operator. 
```

Comment:
this actually solves the database migration problem for us too, which is nice (at least when folks don't have ✨ other tools ✨ to do them out of band of app deployments)